modules/store/src/action_with_modifiers_creator.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  ActionCreator,
+  ActionCreatorProps,
+  NotAllowedCheck,
+  TypedAction,
+} from './models';
+import { createAction } from './action_creator';
+
+export function createActionWithModifiers<
```

Comment:
If this idea is truly worth pursuing, it might be better to make this an overload to `createAction` instead.